### ISSUE TITLE
[Feature]: time_bucket_ng() support OFFSET

### ISSUE THREAD

**Issue by aarondglover** [timescale/timescaledb#4030]
### What problem does the new feature solve?

- OFFSET is already an optional parameter in standard time_bucket() function and feature parity should be sought.
- The reason I use it is to change the "bucketing" behaviour of time_bucket(). The behaviour of time_bucket()/_ng() is to be INCLUSIVE at the Start of the Interval, EXCLUSIVE at END of Interval. I actually need it to be EXCLUSIVE at the beginning of the interval, and INCLUSIVE at the END of the interval. The way that I achieve this is by OFFSET by 1 second.

### What does the feature do?

Allows an offset to be applied to the 'BUCKETING'
In my data - a timestamp at the BEGINNING of an hour (0 minutes, 0 seconds) represents usage within the PRECEDING interval so to achieve my desired outcome I offset the time_bucket by 1 second.

Consider the following data;

```
TS                VAL
--------------------------
00:30:00          1
01:00:00          2
01:30:00          8
02:00:00          3
02:30:00          4
03:00:00          6


SELECT 
  TS, 
  TIME_BUCKET(‘1 Hour’, TS, interval ‘1 sec’) AS BUCKET, 
  VAL 
FROM MYHYPERTABLE;


TS          BUCKET                                          VAL
------------------------------------------------------------------
00:30:00    00:00:01                                        1
01:00:00    00:00:01                                        2
01:30:00    01:00:01                                        8
02:00:00    01:00:01                                        3
02:30:00    02:00:01                                        4
03:00:00    02:00:01                                        6


SELECT 
  TIME_BUCKET(‘1 Hour’, TS, interval ‘1 sec’) AS BUCKET, 
  SUM(VAL) 
FROM MYHYPERTABLE
GROUP BY 
  BUCKET;


BUCKET            SUM(VAL)
--------------------------
00:00:01          3
01:00:01          11
02:00:01          10

```
As an aside - to make the data in my application appear "more normal", for display purposes, I then subtract 1 second from the resultant bucket .
```
SELECT 
TIME_BUCKET(‘1 Hour’, TS, interval ‘1 sec’) – INTERVAL ‘1 sec’ AS BUCKET, 
SUM(VAL) 
FROM DATATABLE
GROUP BY 
BUCKET;


BUCKET            SUM(VAL)
--------------------------
00:00:00          3
01:00:00          11
02:00:00          10
```

### Implementation challenges

OFFSET (as should all other "optional" parameters such as timezone) should be supported in CONTINUOUS AGGREGATES.

**Comment by afiskon** [comment]
@aarondglover Many thanks for this feature request. We already have it on the roadmap and expect it to be shipped somewhere between 2.7.0 and 2.8.0. Stay tuned!

**Comment by svenklemm** [comment]
Should be fixed by #4641
